src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -83,11 +83,14 @@ float64_t RandomFourierDotFeatures::post_dot(float64_t dot_result, index_t par_i
 	return std::cos(dot_result) * constant;
 }
 
-SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
+void RandomFourierDotFeatures::generate_random_parameter_vector(SGVector<float64_t>&& vec)
```

Comment:
do we need this to be rvalue?

@gf712 ?

Reply:
no, should be lvalue ref. rvalue ref implies that you are transferring ownership. In this case you won't see a difference, but this is bad practice, as the caller should now assume that the parameter `vec` has transferred it's data ownership to this function, which I don't think is the case? All you're doing here is mutate `vec`, right?

Reply:
About mutating, you're correct, we're only feeding values to  vec inside the method.

Reply:
@gf712 but get_column would return a temporary value we'll get "cannot bind non const lvalue to rvalue", if we take const lvalue then we can't modify it. So isn't && the only way?

Reply:
I see. Imo you should then store get_column vector in a variable and then pass that to this function.

Reply:
Rather, make it pass-by-value?

Reply:
no, reference is better (this will be called in a loop) so as gil said above